### PR TITLE
Create ansible gendoc

### PR DESCRIPTION
**Pull request by Sail0rd**
## Description of changes

giturlparse: https://github.com/nephila/giturlparse
ansible-gendoc: https://github.com/claranet/ansible-gendoc

## Things done

added giturlparse and ansible-gendoc python packages

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/giturlparse/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ buildPythonPackage
+, fetchPypi
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "giturlparse";
+  version = "0.12.0";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/giturlparse/default.nix:8]:
```suggestion
  version = "0.12.0";

```

File: pkgs/development/python-modules/giturlparse/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "wP/3whrMQ1SRsXeVZuA4dXogXB/9y0fk+B6lKtjDhZo=";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/giturlparse/default.nix:11]:
```suggestion
    hash = "sha256-wP/3whrMQ1SRsXeVZuA4dXogXB/9y0fk+B6lKtjDhZo=";
```

File: pkgs/development/python-modules/ansible-gendoc/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  version = "0.1.11";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "tZj40TzmhPUPANCL48Jsmeyzv8PVFHtsSdKQPkS0b+k=";
+  };
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  propagatedBuildInputs = [
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/ansible-gendoc/default.nix:25]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/ansible-gendoc/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+, giturlparse
+}:
+
+buildPythonPackage rec {
+  pname = "ansible-gendoc";
+  version = "0.1.11";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "tZj40TzmhPUPANCL48Jsmeyzv8PVFHtsSdKQPkS0b+k=";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/ansible-gendoc/default.nix:20]:
```suggestion
    hash = "sha256-tZj40TzmhPUPANCL48Jsmeyzv8PVFHtsSdKQPkS0b+k=";
```

File: pkgs/development/python-modules/ansible-gendoc/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/ansible-gendoc/default.nix:1]: If the package is to be used as command line tool, not python module, it should be placed in pkgs/by-name.

File: pkgs/development/python-modules/ansible-gendoc/default.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    rich
+    pyaml
+    gitpython
+    typer
+    giturlparse
+  ];
+
+  doCheck = false;
+
+  postInstall = ''
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/ansible-gendoc/default.nix:37]: See https://github.com/NixOS/nixpkgs/issues/308283

Comment (Sail0rd) [reply pkgs/development/python-modules/ansible-gendoc/default.nix:37]: Thanks a lot for your feedback, unfortunately I tried to specify the build system and it appears that this package rely on old version of other packages that aren't in nixpkgs and `pythonRelaxDeps` does not seem to work, I think I'll have to close the PR :/ but thanks for taking the time to review this :)

**Pull request closed without merging**